src/scenic/simulators/carla/simulator.py

```diff
@@ -171,6 +176,8 @@ def createObjectInSimulator(self, obj):
 		carlaActor.set_simulate_physics(obj.physics)
 
 		if isinstance(carlaActor, carla.Vehicle):
+			obj.width = carlaActor.bounding_box.extent.y * 2
+			obj.length = carlaActor.bounding_box.extent.x * 2
```

Comment:
Mutating these properties should not be done at simulation time, if possible: Scenic needs to know the dimensions of the objects during scene generation to ensure that no two objects collide, etc. Can the bounding box be extracted earlier? If not, we could leave this in as a hack for now, but having the width and length change between the sampled scene and the generated simulation would be quite surprising for users I think.

Reply:
That's a good point on inconsistency between a scene and its simulation. However, I couldn't find a way to retrieve the size of an actor before it is spawned. I don't think CARLA blueprints have size attributes.

Reply:
Yes, I can't find anything either. We can leave this for now.

Reply:
One way to handle the dimensions would be the following:
Before simulating the Scenic scenario, the Scenic-Carla interface can spawn all the blueprints available in the blueprint library, then measure the dimensions of each agent and filter out those blueprints that don't match the size of any of the the scenario's actors. If no blueprint remains, an error is reported that the simulator cannot support this scenario.